src/abi.h

```diff
@@ -142,6 +143,7 @@ typedef struct authentication_settings {
     char* client_key;
     char* client_key_password;
     char* phase2_auth;  /* netplan-feature: auth-phase2 */
+    char* psk;
```

Comment:
We are passing `NetplanAuthenticationSettings*` to some functions in nm.c, networkd. and nm-parse.c. Moving psk out of it would require changing those function (and possibly parse.c). In the recent past we also added NetplanAuthPMFMode to this struct https://github.com/canonical/netplan/pull/402

I understand it will change the netdef struct layout, but do you it's worth doing that?

Reply:
Hmm #402 is adding a new feature (that will be part of the ABI breaking 1.0 release), so it wasn't really intended for SRU backport and therefore not as critical and a bit of a different story.

We have some guardrails in place in the Ubuntu packaging, like a strict `netplan.io VERSION == libnetplan0 VERSION` requirement. But not all distros might have that, so people could be affected. OTOH, our ABI checker didn't trigger due to this change, so I wonder what's really happening. We should do a more detailed analysis and think this through.

I would actually like to keep it as-is going forward, as this is the much cleaner approach. But when doing a backport/SRU we need to remember to potentially adopt this.